runtime/tutorials/chat_app.md

```diff
@@ -163,73 +140,154 @@ function broadcast_usernames() {
-    broadcast_usernames();
-  };
-
-  // broadcast new message if someone sent one
-  socket.onmessage = (m) => {
-    const data = JSON.parse(m.data);
-    switch (data.event) {
-      case "send-message":
-        broadcast(
-          JSON.stringify({
-            event: "send-message",
-            username: socket.username,
-            message: data.message,
-          }),
-        );
-        break;
-    }
-  };
-});
+};
 
-app.use(router.routes());
-app.use(router.allowedMethods());
-app.use(async (context) => {
-  await context.send({
-    root: `${Deno.cwd()}/`,
-    index: "public/index.html",
+// Update user list in the DOM
+function updateUserList(usernames) {
+  const userList = document.getElementById("users");
+  userList.innerHTML = ""; // Clear existing list
+
+  for (const username of usernames) {
+    const listItem = document.createElement("li");
+    listItem.textContent = username;
+    userList.appendChild(listItem);
+  }
+}
+
+// Add a new message to the conversation
+function addMessage(username, message) {
+  const conversation = document.getElementById("conversation");
+  const messageDiv = document.createElement("div");
+  messageDiv.innerHTML = `<span>${username}</span> ${message}`;
+  conversation.prepend(messageDiv);
+}
+
+// Set up event listener for sending messages
+window.onload = () => {
+  const inputElement = document.getElementById("data");
+  const sendButton = document.getElementById("send");
+
+  // Listen for Enter key press
+  inputElement.addEventListener("keypress", (e) => {
+    if (e.key === "Enter") {
+      const message = inputElement.value;
+      inputElement.value = "";
+      socket.send(JSON.stringify({ event: "send-message", message }));
+    }
   });
-});
 
-console.log("Listening at http://localhost:" + port);
-await app.listen({ port });
+  // Listen for button click
+  sendButton.addEventListener("click", (e) => {
+    e.preventDefault();
+    const message = inputElement.value;
+    inputElement.value = "";
+    socket.send(JSON.stringify({ event: "send-message", message }));
+  });
+};
 ```
 
-We can start our server with the following command. Note we need to explicitly
-grant access to the file system and network because Deno is secure by default.
+This code sets up a WebSocket connection to the server and listens for messages
+from the server. When a new message is received, if it is a `send-message` event
+it updates the DOM with the new message, and if it is an `update-users` event it
+updates the list of active users. It also sets up event listeners for either
+pressing the Enter key or clicking the send button to send a message to the
+server.
 
-```sh
-deno run --allow-read --allow-net server.js
+## Running the server
+
+To run the server we'll need to grant the necessary permissions to Deno. In your
+`deno.json` file, update the `dev` task to allow read and network access:
+
+```sh title="deno.json"
+"dev": "deno run --allow-net --allow-read --watch main.ts"
```

Comment:
Maybe we should do a diff here to see what's changed compared to the old command.